hermeto/core/package_managers/maven/main.py

```
@@ -0,0 +1,554 @@
+
+    @classmethod
+    def from_file(cls, lockfile_path: RootedPath) -> "MavenLockfile":
+        """Create a MavenLockfile from a lockfile.json file."""
+        try:
+            with lockfile_path.path.open("r") as f:
+                lockfile_data = json.load(f)
+        except FileNotFoundError:
+            raise PackageRejected(
+                f"The {DEFAULT_LOCKFILE_NAME} file must be present for the maven package manager",
+                solution=f"Please ensure that {DEFAULT_LOCKFILE_NAME} is present in the package directory",
+            )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)